pkg/metrics/metrics.go

```diff
@@ -0,0 +1,33 @@
+package metrics
+
+import "github.com/prometheus/client_golang/prometheus"
+
+var (
+	registry          = prometheus.NewRegistry()
```

Comment:
I recommend to not introduce yet another global registry, but rather pass it around to where it's needed

Reply:
What do you mean? This is not global nor exported, it is part of a package and it is impossible to use it anywhere else. It is declared where it is used and there is no reason to pass it around. Could you clarify?

Reply:
it's still package global, which is a bit better but function scoped is better in my opinion :)

Reply:
I see. Correct me if I am wrong but you are advocating in favor of something closer to a dependency injection approach, where I would inject this "registry" in other entities(packages or structs) that then would register their own "counters" and "metrics" towards it. It has its charm i admit, and that was my initial approach. That being said, I advocate for a less intrusive approach: make metrics centralized and accessible through exported functions. :)

Reply:
it ultimately comes down to preference, we've had bad experiences with anything other than dependency injection (it also allows you to use the [label/name-prefix currying](https://github.com/prometheus/client_golang/blob/20428fa0bffcb4ee7a7941545ae0df9acdc9d8a7/prometheus/wrap.go#L25-L47))